cluster-api-gcp/cloud/google/machineactuator.go

```diff
@@ -673,3 +690,33 @@ func getSubnet(netRange clusterv1.NetworkRanges) string {
 	}
 	return netRange.CIDRBlocks[0]
 }
+
+// Takes a Machine object representing the current state, and update it based on the actual
+// current state as reported by GCE, kubelet etc.
+func currentMachineActualStatus(gce *GCEClient, currentMachine *clusterv1.Machine) (*clusterv1.Machine, error) {
+	instance, err := gce.instanceIfExists(currentMachine)
+	if err != nil {
+		return nil, err
+	}
+
+	machineType := strings.Split(instance.MachineType, "/")
+	zone := strings.Split(instance.Zone, "/")
+
+	config, e := gce.providerconfig(currentMachine.Spec.ProviderConfig)
+	if e != nil {
+		return nil, apierrors.InvalidMachineConfiguration("Cannot unmarshal providerConfig field: %v", e)
+	}
+
+	config.MachineType = machineType[len(machineType)-1]
+	config.Zone = zone[len(zone)-1]
+
+	b := []byte{}
```

Comment:
Would be nice to have the encode in a helper function to match the decode function for the provider config. Not required for this review though.